docker/configure_workers_and_start.py

```diff
@@ -389,29 +565,232 @@ def generate_base_homeserver_config() -> None:
+
+    # The final result of all this processing
+    dict_to_return: Dict[str, Any] = {}
+
+    # Handle any multipliers requested for a given worker.
+    multiple_processed_worker_types = apply_requested_multiplier_for_worker(
+        requested_worker_types
+    )
+
+    # Check each requested worker_type_string for a user-requested name and parse it
+    # out if present.
+    for worker_type_string in multiple_processed_worker_types:
+        requested_worker_name = ""
+        new_worker_type_string = worker_type_string
+        # First, check if a name is requested
+        if "=" in worker_type_string:
+            # Split on "=", remove extra whitespace from ends then make list
+            worker_type_split = split_and_strip_string(worker_type_string, "=")
+            if len(worker_type_split) > 2:
+                error(
+                    "To many worker names requested for a single worker, or to many "
+                    f"'='. Please fix: {worker_type_string}"
+                )
+            # Assign the name
+            # Hopefully, no one will actually use whitespace in a requested worker name,
+            # but if they do, replace it with an underscore to maintain readability.
+            requested_worker_name = worker_type_split[0].replace(" ", "_")
+
+            # If there was no name given, this will still be an empty string. If we got
+            # here, it's because there was a name requested with a '='. They must have
+            # forgotten to actually type in a name. e.g. '=type'
+            if not requested_worker_name:
+                error(
+                    "Worker name requested but not actually found at: "
+                    f"'{worker_type_string}'. Please fix."
+                )
+
+            # Check the last character of a requested name is not a number. This can
+            # cause an error that comes across during startup as an exception in
+            # 'startListening' and ends with 'Address already in use' for the port.
+            # This only seems to occur when requesting more than 10 of a given
+            # worker_type, otherwise it would be ok. Check this separately from the
+            # regex below, as otherwise digits are ok in a name
+            if requested_worker_name[-1].isdigit():
+                error(
+                    "Found a number at the end of the requested worker name: "
+                    f"{requested_worker_name}. This is not allowed as it will cause "
+                    "exceptions with 'Address already in use'. Recommend appending an "
+                    "underscore after the number if this what you really want to do."
+                )
+
+            # Uncommon mistake that will cause problems. Name string containing any of a
+            # range of nasties will do Bad Things to filenames and probably nginx too.
+            if not re.match(r"^[a-zA-Z0-9_+-]*[a-zA-Z0-9_+-]$", requested_worker_name):
```

Comment:
```suggestion
            if not re.match(r"^[a-zA-Z0-9_+-]*[a-zA-Z_+-]$", requested_worker_name):
```
I believe the intention was also to not let you finish the worker name with a number